authz/providers/azure/rbac/rbac.go

```diff
@@ -517,7 +517,8 @@ func (a *AccessInfo) sendCheckAccessRequest(ctx context.Context, checkAccessUser
 	}
 
 	klog.V(7).Infof("checkaccess response: %s, Configured ARM call limit: %d", string(data), a.armCallLimit)
-	if resp.StatusCode != http.StatusOK {
+	// We can expect the response to be a 200 OK for proxy resources or 404 Not Found for non-proxy resources due to resource deletion
```

Comment:
```suggestion
	// We can expect the response to be a 200 OK for ARM proxy resources or 404 Not Found for ARM tracked resources due to resource deletion
```
i suggest using ARM terminology here as "proxy" can be interpreted under different context.